iceoryx2-bb/posix/src/periodic_timer.rs

```diff
@@ -0,0 +1,235 @@
+    pub fn reset(&self) -> Result<(), TimeError> {
+        self.periodic_timer.reset(self.index)
+    }
+}
+
+impl<'periodic_timer> Drop for PeriodicTimerGuard<'periodic_timer> {
+    fn drop(&mut self) {
+        self.periodic_timer.remove(self.index);
+    }
+}
+
+/// Builder to create a [`PeriodicTimer`].
+pub struct PeriodicTimerBuilder {
+    clock_type: ClockType,
+}
+
+impl Default for PeriodicTimerBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl PeriodicTimerBuilder {
+    /// Creates a new builder.
+    pub fn new() -> Self {
+        Self {
+            clock_type: ClockType::default(),
+        }
+    }
+
+    /// Defines the [`ClockType`] that is used for time measurements. By default it is
+    /// [`ClockType::default()`].
+    pub fn clock_type(mut self, value: ClockType) -> Self {
+        self.clock_type = value;
+        self
+    }
+
+    /// Creates a new [`PeriodicTimer`]
+    pub fn create(self) -> Result<PeriodicTimer, TimeError> {
+        let start_time = fail!(from "PeriodicTimer::new()", when Time::now_with_clock(self.clock_type),
+                                "Failed to create PeriodicTimer since the current time could not be acquired.");
+        let start_time = start_time.as_duration().as_nanos();
+
+        Ok(PeriodicTimer {
+            timeouts: RefCell::new(vec![]),
+            id_count: IoxAtomicU64::new(0),
+            clock_type: self.clock_type,
+            last_iteration: RefCell::new(start_time),
+        })
+    }
+}
+
+#[derive(Debug)]
+struct Attachment {
+    index: u64,
+    period: u128,
+    start_time: u128,
+}
+
+impl Attachment {
+    fn new(index: u64, period: u128, clock_type: ClockType) -> Result<Self, TimeError> {
+        let start_time = fail!(from "Attachment::new()", when Time::now_with_clock(clock_type),
+                                "Failed to create PeriodicTimer attachment since the current time could not be acquired.");
+        let start_time = start_time.as_duration().as_nanos();
+
+        Ok(Self {
+            index,
+            period,
+            start_time,
+        })
+    }
+
+    fn reset(&mut self, clock_type: ClockType) -> Result<(), TimeError> {
+        let start_time = fail!(from "Attachment::new()", when Time::now_with_clock(clock_type),
+                                "Failed to reset PeriodicTimer attachment since the current time could not be acquired.");
+        self.start_time = start_time.as_duration().as_nanos();
+        Ok(())
+    }
+}
+
+/// The [`PeriodicTimer`] allows the user to attach multiple periodic timers with
+/// [`PeriodicTimer::add()`], to wait on them by acquiring the waiting time to the next timer
+/// with [`PeriodicTimer::next_iteration()`] and to acquire all missed timers via
+/// [`PeriodicTimer::missed_timers()`].
+#[derive(Debug)]
+pub struct PeriodicTimer {
+    timeouts: RefCell<Vec<Attachment>>,
+    id_count: IoxAtomicU64,
+    last_iteration: RefCell<u128>,
```

Comment:
Why "iteration" ? Naming is inconsistent.

Also, prefer `previous` to `last` to avoid ambiguity.

Reply:
iterations relates to when the last `next_iteration` call was called. I have no better name for this.